src/compiler/checker.ts

```diff
@@ -22576,6 +22576,15 @@ namespace ts {
             let hasComputedNumberProperty = false;
             propertiesTable = createSymbolTable();
 
+            // Spreads may cause an early bail; ensure computed names are always checked (this is cached)
+            // As otherwise they may not be checked until exports for the type at this position are retrieved,
+            // which may never occur.
+            for (const elem of node.properties) {

```

Comment:
Why do you need to iterate here explicityly and not do it as part of the loop below.

Reply:
Got it .. Just saw your comment about the return expression